packages/hardhat/contracts/Ethernauts.sol

```diff
@@ -302,12 +301,10 @@ contract Ethernauts is ERC721Enumerable, Ownable {
 
         _mint(to, tokenId);
 
-        uint256 currentBatchId = _randomNumbers.length;
-        uint256 maxTokenIdInBatch = batchSize * (currentBatchId + 1) - 2;
+        uint256 currentBatchId = tokenId / batchSize;
+        uint256 maxTokenIdInBatch = batchSize * (currentBatchId + 1) - 1;
 
         if (tokenId == maxTokenIdInBatch) {
-            emit BatchEnd(currentBatchId);
```

Comment:
I'm removing the `BatchEnd` event, as the logic to check if the current batch has ended is going to be moved to the JS logic, so we can have a leaner contract.

Reply:
That makes sense

Reply:
I assume once the batch is minted, we upload to ipfs, call setUrl, right?

Reply:
Yes